libs/community/langchain_community/graph_vectorstores/extractors/link_extractor.py

```diff
@@ -14,7 +14,7 @@ class LinkExtractor(ABC, Generic[InputT]):
     """Interface for extracting links (incoming, outgoing, bidirectional)."""
 
     @abstractmethod
-    def extract_one(self, input: InputT) -> set[Link]:  # noqa: A002
+    def extract_one(self, input: InputT) -> Set[Link]:
```

Comment:
Why does the API use a Set vs. a Link?

1) Will all links be hashable?
2) Is order of appearance of results never valuable?

Reply:
1. All links are hashable
2. Order of appearance is not valuable in the graph -- a node either has a link or it doesn't.

I could see a case for `List` (preserving the order of hyperlinks in an HTML document, for instance), but the order wouldn't have any affect on traversal (eg., once you write the links into a DB and get them back, the order would be lost).

Basically -- links induce edges, and an edge exist (is present) or not.